src/main/java/com/lgtm/easymoney/services/impl/FriendServiceImpl.java

```diff
@@ -0,0 +1,172 @@
+package com.lgtm.easymoney.services.impl;
+
+import com.lgtm.easymoney.enums.UserType;
+import com.lgtm.easymoney.exceptions.InapplicableOperationException;
+import com.lgtm.easymoney.exceptions.InvalidUpdateException;
+import com.lgtm.easymoney.exceptions.ResourceNotFoundException;
+import com.lgtm.easymoney.models.Friendship;
+import com.lgtm.easymoney.models.User;
+import com.lgtm.easymoney.payload.FriendshipReq;
+import com.lgtm.easymoney.payload.ProfileRsp;
+import com.lgtm.easymoney.payload.ProfilesRsp;
+import com.lgtm.easymoney.repositories.FriendshipRepository;
+import com.lgtm.easymoney.services.FriendService;
+import com.lgtm.easymoney.services.UserService;
+import java.util.ArrayList;
+import java.util.List;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+
+/**
+ * service for friendship.
+ * TODO after we have auth, move the inapplicable op checks to a HandlerInterceptor instead
+ */
+@Service
+@Transactional(rollbackFor = Exception.class)  // required for delete
+public class FriendServiceImpl implements FriendService {
+  private final UserService userService;
+
+  @Autowired
+  private final FriendshipRepository friendshipRepository;
+
+  @Autowired
+  public FriendServiceImpl(UserService userService, FriendshipRepository friendshipRepository) {
+    this.userService = userService;
+    this.friendshipRepository = friendshipRepository;
+  }
+
+  @Override
+  public Friendship getFriendshipRecord(User u1, User u2) {
+    return friendshipRepository.findByUser1AndUser2(u1, u2);
+  }
+
+  @Override
+  public void addFriend(FriendshipReq friendshipReq) {
+    User requester = userService.getUserById(friendshipReq.getUid1());
+    if (requester.getType() != UserType.PERSONAL) {
```

Comment:
lets create a private method in  `friendservice` for checking if the type is allowed, since friend feature is open available to end users.

Reply:
I left a TODO comment on top.